--- pango-1.32.0/pango/Makefile.am.lib64	2012-09-27 04:35:16.000000000 +0800
+++ pango-1.32.0/pango/Makefile.am	2012-09-27 09:24:38.000000000 +0800
@@ -17,6 +17,7 @@
 	-DPANGO_ENABLE_BACKEND				\
 	-DPANGO_ENABLE_ENGINE				\
 	-DSYSCONFDIR=\"$(sysconfdir)\"			\
+	-DARCH=\"$(ARCH)\"				\
 	-DLIBDIR=\"$(libdir)\"				\
 	-I$(top_srcdir)					\
 	-I$(top_builddir)				\
--- pango-1.32.0/pango/modules.c.lib64	2012-09-15 10:11:17.000000000 +0800
+++ pango-1.32.0/pango/modules.c	2012-09-27 09:26:34.000000000 +0800
@@ -549,6 +549,7 @@
       files = g_new (char *, 3);
 
       files[0] = g_build_filename (pango_get_sysconf_subdirectory (),
+		      		   ARCH,
                                    "pango.modules",
                                    NULL);
       files[1] = g_build_filename (pango_get_lib_subdirectory (),
@@ -668,6 +669,7 @@
       if (!no_module_warning)
 	{
 	  gchar *filename = g_build_filename (pango_get_sysconf_subdirectory (),
+					      ARCH,
 					      "pango.modules",
 					      NULL);
 	  g_critical ("No modules found:\n"
